### DISCUSSION TITLE
Showing city names in addition to countries

### DISCUSSION THREAD

**Discussion by torhus** [torhus/monster-browser#64] [Ideas]
I just tried this. Not sure how useful it is, given that you already have the ping value. One problem is that the IP-to-location database file for this is 98 MB...

![mb_city](https://user-images.githubusercontent.com/19881605/204679319-7730a210-d683-4315-acd0-23870f7997dc.png)

**Comment by The-Gig** [comment]
Uhm... apart from the fact that IP-City associations aren't aways right, the size of the database is various times bigger than the rest of the program.
If you like it, I would think as an in-app dowload, like you open the Tools menu and find a "Download/Update Cities Database" option.

**Reply by The-Gig** [reply]
... possibly explaining in instructions which one is the file to delete in case people then change their mind and want to shrink their MB install.

**Reply by torhus** [reply]
I think I could have this as a documented option for people that want to use it. It only requires checking for city information and adding it when it's there, and a setting to customize the database file name. There is a build here:
https://ci.appveyor.com/project/torhus/monster-browser/builds/45528543/artifacts

The precision will vary depending on which database you use, I used the free DB-IP IP to City Lite database now. It will give you cities for all addresses, while GeoIP City Lite while give you about half, I assume the ones that are most likely to be correct.

**Reply by The-Gig** [reply]
I downloaded the Nov 30 2022 build, but maybe I'm missing something, as the only change I noticed is that in settings.ini there is a new line "geoIpDatabase" to set the name of the ip database in use... I don't see options nor documentation which tell the user _where_ he can get a more extensive database which includes cities.

By the way, I only now noticed that it's possible to enlarge the first column to show the name of the nation: up until now I thought it only included the flag and one needed to mouseover to read the name of the country. _Wondering whether that column should be made larger by default..._

**Reply by torhus** [reply]
> I downloaded the Nov 30 2022 build, but maybe I'm missing something, as the only change I noticed is that in settings.ini there is a new line "geoIpDatabase" to set the name of the ip database in use... I don't see options nor documentation which tell the user _where_ he can get a more extensive database which includes cities.

It's not documented currently, but you can download the MMDB file from here if you want to try it:
https://db-ip.com/db/download/ip-to-city-lite

> 
> By the way, I only now noticed that it's possible to enlarge the first column to show the name of the nation: up until now I thought it only included the flag and one needed to mouseover to read the name of the country. _Wondering whether that column should be made larger by default..._

The name is just there to get the tool tip, I assume that people recognize the flags of nearby countries anyway :)

**Reply by The-Gig** [reply]
I download the .gz, extracted the .mmdb from it and then edited settings.ini. Various (relatively simple) steps required, then works like a charm.
